File: slack-reader/src/main/java/dev/number6/slackreader/adaptor/EnvironmentVariableSlackReaderConfigurationAdapter.java
package dev.number6.slackreader.adaptor;

import dev.number6.slackreader.port.SlackReaderConfigurationPort;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collection;

public class EnvironmentVariableSlackReaderConfigurationAdapter implements SlackReaderConfigurationPort {

    private static final String SLACK_MSG_TOPIC_ARN = "SLACK_MSG_TOPIC_ARN";
    private static final String BLACKLISTED_CHANNELS = "BLACKLISTED_CHANNELS";

    private final String topicArn = System.getenv(SLACK_MSG_TOPIC_ARN);
    private final String blackListedChannels = System.getenv(BLACKLISTED_CHANNELS);

    @Override
    public String getTopicArn() {
        return topicArn;
    }

    @Override
    public Collection<String> getBlacklistedChannels() {
        return blackListedChannels == null ? new ArrayList<>() : Arrays.asList(blackListedChannels.split(","));
    }
}

File: slack-reader/src/main/kotlin/dev/number6/slackreader/adaptor/EnvironmentVariableSlackReaderConfigurationAdapter.kt
package dev.number6.slackreader.adaptor

import dev.number6.slackreader.port.SlackReaderConfigurationPort

class EnvironmentVariableSlackReaderConfigurationAdapter : SlackReaderConfigurationPort {
    private val topicArn = System.getenv(SLACK_MSG_TOPIC_ARN)
    private val blackListedChannels = System.getenv(BLACKLISTED_CHANNELS)
    override fun getTopicArn(): String {
        return topicArn
    }

    override fun getBlacklistedChannels(): Collection<String> {
        return if (blackListedChannels == null) listOf() else listOf(*blackListedChannels.split(",".toRegex()).toTypedArray())
    }

    companion object {
        private val SLACK_MSG_TOPIC_ARN: String? = "SLACK_MSG_TOPIC_ARN"
        private val BLACKLISTED_CHANNELS: String? = "BLACKLISTED_CHANNELS"
    }
}